### DISCUSSION TITLE
Some Curator integration questions

### DISCUSSION THREAD

**Discussion by boris-glumpler** [awcodes/filament-curator#488] [Q&A] [answered]
Hello Adam,

I'm working on paste/drop uploads for my Milkdown plugin atm and decided to already integrate Curator now and I'm hoping you can help me understand a couple things.

I had a look at the `creating` event in the model observer and it looks like quite a few attributes are being set there, but what exactly is the shape of `$media->file`? I also didn't find where it's being set, so couldn't track it down.

Then, I'm not sure how `curations` are saved on the model? Should I just save all the default presets here or is that column just used for custom curations that are not default presets?

I think these two things are the last pieces of the puzzle for me and then this part of the integration should be good to go.

Thanks for your time!
~Boris

**Comment by awcodes** [comment, marked as answer]
Hey Boris,

`file` is coming from the saveUploadedFiles() method in `src\Components\Forms\Uploader.php`. it's an override to the native filament upload.

As for the curations, those are stored as a json on the curations db field. 

Curator takes a different approach to handling media than we are traditionally used to. Typically you would create multiple sizes of images during the upload and storing them all to the disk. This creates a lot of files that might not ever get used and just sit there taking up space on your server for no reason. Curator's approach is to upload just the one asset and serve images, etc through Glide which dynamically creates the actual needed sizes and caching them before serving the image back out to the browser.

So, curations are intended to be used to force focal points for specific images that may not crop right at set sizes, like thumbnails. but curations are a manual process since there's no way programmatically guarantee the focal point without seeing it. You, certainly could generate them, but it's not necessary.

**Reply by boris-glumpler** [reply]
Ok, got it! So, since I set all the attributes manually already and curations might be provided later on by the user, I should be all set. Thanks, Adam!
